contracts/JBController.sol

```diff
@@ -825,20 +873,36 @@ contract JBController is IJBController, JBTerminalUtility, JBOperatable, Reentra
       JBFundAccessConstraints memory _constraints = _fundAccessConstraints[_i];
 
       // Set the distribution limit if there is one.
-      if (_constraints.distributionLimit > 0)
-        distributionLimitOf[_projectId][_fundingCycle.configuration][
+      if (_constraints.distributionLimit > 0) {
+        // The distribution limit should fit in a uint248.
+        require(_constraints.distributionLimit < type(uint248).max, '0x00: BAD_DISTRIBUTION_LIMIT');
+
+        // The currency of the distribution limit should fit in a uint8.
+        require(
+          _constraints.distributionLimitCurrency < type(uint8).max,
+          '0x00: BAD_DISTRIBUTION_LIMIT_CURRENCY'
+        );
+
+        _packedDistributionLimitDataOf[_projectId][_fundingCycle.configuration][
           _constraints.terminal
-        ] = _constraints.distributionLimit;
+        ] = _constraints.distributionLimit | (_constraints.distributionLimitCurrency << 248);
+      }
 
       // Set the overflow allowance if there is one.
-      if (_constraints.overflowAllowance > 0)
-        overflowAllowanceOf[_projectId][_fundingCycle.configuration][
-          _constraints.terminal
-        ] = _constraints.overflowAllowance;
+      if (_constraints.overflowAllowance > 0) {
+        // The currency of the overflow allowance should fit in a uint248.
+        require(_constraints.overflowAllowance < type(uint248).max, '0x00: BAD_OVERFLOW_ALLOWANCE');
+
+        // The currency of the overflow allowance should fit in a uint8.
+        require(
+          _constraints.overflowAllowanceCurrency < type(uint8).max,
+          'BAD_OVERFLOW_ALLOWANCE_CURRENCY'
```

Comment:
We should use '0x00: BAD_OVERFLOW_ALLOWANCE_CURRENCY' until the custom errors from @viraj124 come in

Reply:
nice. done.